.github/workflows/autotest.yml

```diff
@@ -5,7 +5,25 @@ on:
   - push
 
 jobs:
+  header-lint:
+    if: 'github.event_name == ''pull_request'''
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - run: |
+          git pull origin master
+          echo "Current git rev $(git rev-parse HEAD)"
+          echo "Git rev $(git merge-base HEAD remotes/origin/master)"
+          echo "If a file header needs to be changed run "tools/add_header -i FILE" from the rucio project directory."
+          for file in $(git diff --name-status HEAD $(git merge-base HEAD remotes/origin/master) | grep -v '^A' | cut -f 2 | grep -E '\.py$|^bin/'); do
+            echo "Processing $file"
+            tools/add_header --disable-add-me -cd $file
+          done
   setup:
+    if: ${{ success() || cancelled() }}
```

Comment:
Umm, why is `success() || cancelled()`required? I assume that it is equivalent to `jobs.header-lint.status`, in which case, we can mention that explicitly ;) 

In the current setup, if `header-lint` job fails, the tests will not run at all. 

I guess for both `setup` and `release-patch-setup` jobs we can remove the `success() or failfure()` and `needs: header-lint` parts.

I'd probably let the test suite run even if the `header-lint` job fails. ;)

Reply:
I see a logical flaw in it:
The developer pushed some changes. If the headers are not correct, the tests will run nevertheless. However, since the tests are not correct, the developer has to update the headers and push the changes again. This means, that the tests will run twice with essentially the same code (just the headers changed). Do we want that?

In the current case, the tests early abort if the headers are not correct, this normally happens after 1 minute or less.

Reply:
In the spirit of discovering all flaws with a PR early, we'd definitely want that ;) 

Diving into some details now,

There will be a new entry in every PR which would look like this:
```
Run all the autotests / header-lint (pull_request)
```
regardless of whether we choose to make `header-lint` a hard dependency on the following jobs (setup, test, ..). So, if the linting fails, the CI will not be in a 'green, ready to merge' state in either case.
I was thinking on the lines of exposing all the issues with a PR upfront i.e header lints and failing tests.

Let's consider the following cases:
1) headers are incorrect and tests are failing
If `header-lint` is not a hard dependency, the developer will be made aware of both the incorrect headers and the failing tests in the same GitHub Action run. You generally see all the jobs and success/failure statistics in the PR.
So the developer has a chance to fix both headers and failed tests in the next push. 
Otherwise, he/she would first fix the headers and pushes the code, then lets the tests run and discover the failing tests and then deal with the failing tests.

2) headers are correct and tests are failing
No problem here in either case i.e if header-lint is a hard dependency or not. 

3) headers are incorrect and tests are passing
In this case, the PR would show `Some checks were not successful` indicating that header-linting failed.

4) headers are correct and tests are passing
All good here in either case.

So I vote for keeping `header-lint` as an independent check for PRs, to allow discovery of test errors early in the PR pipeline.

Let me know what you think ;) 

---
Whether or not we choose to keep the dependency, the following line is redundant
```
 if: ${{ success() || cancelled() }}
```
since it is already accommodated by
```
needs: header-lint
```
Since debugging Github Actions can be a pain, I have simulated a simple example to demo this:
See the code change:
https://github.com/maany/rucio/commit/494db8fd824fc29db51b574ec9f92aa5844cceb2 
and the corresponding run:
https://github.com/maany/rucio/actions/runs/1707477530 

The run fails if `header-lint` fails despite dropping the `if` condition.


-----

Reply:
Okay, I'll change it. For me, there is no strong reason to make it a dependency.

FYI The
```
 if: ${{ success() || cancelled() }}
```
is mandatory, since the `header-lint` step has the condition
```
if: 'github.event_name == ''pull_request'''
```
. So if the action run is not a PR (e.g. a normal commit) the `header-lint` gets canceled and thus the tests wouldn't have run too.

Since we don't want it to be a dependency any more it is not important.

Reply:
> ```
>  if: ${{ success() || cancelled() }}
> ```
> 
> 
>     
>   
> 
> is mandatory, since the `header-lint` step has the condition
> 
> 
> ```
> if: 'github.event_name == ''pull_request'''
> ```

Aaahh right! maketh sense ;)